openshift/system/Dockerfile

```diff
@@ -1,82 +1,75 @@
-FROM quay.io/centos7/ruby-27-centos7 AS porta-base
-
-USER root
-
+FROM quay.io/centos/centos:stream8 as builder
+WORKDIR /opt/system
 ENV RUBY_MAJOR_VERSION=2 \
-    RUBY_MINOR_VERSION=7
+    RUBY_MINOR_VERSION=7 \
+    RAILS_ENV=production \
+    NODE_ENV=production
+ENV RUBY_VERSION="${RUBY_MAJOR_VERSION}.${RUBY_MINOR_VERSION}"
+ADD . ./
+ADD openshift/system/config/* ./config/
+RUN dnf -y module enable ruby:${RUBY_VERSION} nodejs:16 mysql:8.0 \
+    && dnf install -y --setopt=skip_missing_names_on_install=False,tsflags=nodocs shared-mime-info make automake gcc gcc-c++ redhat-rpm-config postgresql rubygem-irb rubygem-rdoc ruby-devel nodejs libpq-devel mysql-devel gd-devel git 'dnf-command(download)' \
+    && gem install bundler --version 2.2.25 \
+    && bundle config set --local without development:test:licenses \
+    && bundle config set --local deployment true \
+    && bundle install --jobs $(grep -c processor /proc/cpuinfo) --retry=5 \
+    && npm install -g yarn \
+    && yarn install:safe --no-progress \
+    && SECRET_KEY_BASE=rails/32947 bundle exec rake assets:precompile tmp:clear \
+    && rm -rf tmp/cache /usr/local/share/gems/cache /usr/local/share/gems/doc
 
-ENV RUBY_VERSION="${RUBY_MAJOR_VERSION}.${RUBY_MINOR_VERSION}" \
-    RUBY_SCL_NAME_VERSION="${RUBY_MAJOR_VERSION}${RUBY_MINOR_VERSION}"
+# can just install memkind once it is fixed, see
+# https://issues.redhat.com/browse/RHEL-14497
+RUN cd /tmp \
+    && dnf download --source memkind \
+    && rpm2cpio memkind-*.src.rpm | cpio -idmv "memkind-*.tar.gz" \
+    && tar xvfz memkind-*.tar.gz \
+    && cd memkind-*/jemalloc/ \
+    && ./autogen.sh && ./configure --libdir=/usr/local/lib64/ && make
```

Comment:
Just curious... don't you need to call `make install`. Is `/tmp/` the final location of this binary?

Reply:
This is just the builder stage. It doesn't have to be installed. In the last stage the library is copied to the proper location.
I hope that we will not have to compile from sources soon. See https://issues.redhat.com/browse/RHEL-14497